doc/reference/files/dune-project/dialect.rst

```diff
@@ -0,0 +1,71 @@
+dialect
+-------
+
+.. dune:stanza:: dialect
+
+   Declare a new :term:`dialect`.
+
+   .. dune:field:: name
+      :param: <name>
+
+      The name of the dialect being defined. It must be unique in a given
+      project.
+
+      This field is required.
+
+   .. dune:field:: implementation
+
+      Details related to the implementation files (corresponding to `*.ml`).
+
+      .. versionchanged:: 3.9 This field is made optional.
+
+      .. dune:field:: extension
+         :param: <string>
+
+         Specify the file extension used for this dialect.
+
+         The extension string must not start with a period and be unique in a
+         given project (so that a given extension can be mapped back to a
+         corresponding dialect). In Dune 3.9 and later, the extension string may
+         contain periods (e.g., `cppo.ml`).
+
+         This field is required.
+
+      .. dune:field:: preprocess
+         :param: <action>
+
+         Run `<action>` to produce a valid OCaml abstract syntax tree.
+
+         This action is expected to read the file given in the variable named
+         ``%{input-file}`` and output a *binary* abstract syntax tree on its
+         standard output.
+
+         If the field is not present, it is assumed that the corresponding
+         source code is already valid OCaml code and can be passed to the OCaml
+         compiler as-is.
+
+         .. seealso:: :ref:`preprocessing-actions`
+
+      .. dune:field:: format
+         :param: <action>
+
+         Run `<action>` to format source code for this dialect.
+
+         The action is expected to read the file given in the variable named
+         ``%{input-file}`` and output the formatted source code on its standard
+         output.
+
+         If the field is not present, the behaviour depends on the presence of
```

Comment:
```suggestion
         If the field is not present, the behavior depends on the presence of
```
If we're sticking with American spelling in Dune docs (as is my understanding), this should be without the `u`